cordova-lib/src/cordova/plugin.js

```diff
@@ -414,13 +414,15 @@ function determinePluginTarget(projectRoot, cfg, target, fetchOptions) {
     // it does not satisfy.
     if(parsedSpec.version) {
         if(pkgJson && pkgJson.dependencies && pkgJson.dependencies[parsedSpec.package]) {
-            var noSymbolVersion;
+            var noSymbolVersion = parsedSpec.version;
             if (parsedSpec.version.charAt(0) === '^' || parsedSpec.version.charAt(0) === '~') {
                 noSymbolVersion = parsedSpec.version.slice(1);
             }
-            if (!semver.satisfies(noSymbolVersion, pkgJson.dependencies[parsedSpec.package])) {
+            if (pkgJson.dependencies[parsedSpec.package] === parsedSpec.version) {
```

Comment:
I think we still want to do the semver check too. First start with outer if that checks if `parsedSpec.version` is a directory or url,  if it is, do this check  to see if `pkgJson.dependencies[parsedSpec.package] !== parsedSpec.version`. If that is true, then you want to update package.json (assuming fetchOpts.save === true)

On that outer if that checks if it is a dir or url, add a else if where you check for semver.satisfies